src/Service/Connector/AbstractProvider.php

```diff
@@ -55,6 +53,11 @@ public function isSupported(Domain ...$domainList): bool
         return true;
     }
 
+    public function authenticate(array $authData): void
+    {
+        $this->authData = $authData;
```

Comment:
I'm thinking of making this method also call verifyAuthData, thereby making verifyAuthData protected and local.
This can allow us to better us DI and the already created client in the local scopes of providers.

As of right now, it's not possible to introspect the authentication in the Namecheap provider as we don't have access to the $outgoingIp injection within verifyAuthData.

RFC @maelgangloff

Reply:
After discussing it together, I think this is a good idea.
In any case, this is a breaking change so we will have to adapt the other classes that inherit from `AbstractProvider` as well as PR #52 when it is ready to be merged.

Thanks for this thought

Reply:
This PR already has a lot of breaking changes anyway. The autodns PR will have to go before this one.